lib/transaction.js

```diff
@@ -234,6 +235,23 @@ class Transaction extends EventEmitter {
         }
       });
   }
+
+  then(onResolve, onReject) {
+    return this._promise.then(onResolve, onReject);
+  }
+
+  catch(onReject) {
+    return this._promise.catch(onReject);
+  }
+
+  finally(onFinally) {
+    return this._promise.finally(onFinally);
+  }
+
+  asCallback(cb) {
+    callbackify(() => this._promise)(cb);
+    return this._promise;
+  }
```

Comment:
Hmm... it looks like `.asCallback(..)` is mostly a Bluebird method.  Unfortunately, we mentioned it in the Knex documentation, and therefore we (probably) need to continue supporting it.

However, it looks like Bluebird supports an additional `options` parameter:

https://github.com/petkaantonov/bluebird/blob/master/src/nodeify.js#L45

So, even if we continue providing an `.asCallback(..)` method, there's still a good chance that we're gonna break backwards compatibility by introducing this change.

@kibertoad + @maximelkin : Thoughts?

Reply:
We not documented options neither on site nor typings
as solution we can assert second parameter is undefined

Reply:
Then we can freely consider options entity not a part of public api and remove it without semver major.

Reply:
I vote for bump semver major (second number of course), just to warn users

Reply:
It's always a hassle, because then downstream (such as bookshelf) need to bump their versions as well, and in this case it feels hardly warranted.
I would reserve (semi-)major version bump for when we drop Node 8 support.

Reply:
My guess is that this release is prolly gonna surprise end-users one way or another.  There's a good chance that their code probably relied upon `bluebird` methods even though `knex` didn't explicitly advertise support for them.